File: gherkin/dotnet/Gherkin/CucumberMessages/Types/Source.cs
using System.Runtime.Serialization;

namespace Gherkin.CucumberMessages.Types
{
    public class Source
    {
        [DataMember(Name = "uri")]
        public string Uri { get; set; }

        [DataMember(Name = "data")]
        public string Data { get; set; }

        [DataMember(Name = "mediaType")]
        public string MediaType { get; set; }
    }
}

File: gherkin/dart/lib/src/messages/Source.dart
import 'package:gherkin/core.dart';
import 'package:gherkin/extensions.dart';
import 'package:gherkin/messages.dart';

class Source implements INullSafetyObject
{
  static const empty = _InvalidSource();

  final String uri;

  final String data;

  final MediaType mediaType;

  const Source(this.uri, this.data, this.mediaType);

  @override
  bool get isEmpty => false;

  @override
  bool get isNotEmpty => !isEmpty;
}

/// Convenience implementation of an invalid [Source] instance.
class _InvalidSource
    extends Source
{
  const _InvalidSource()
      : super(Strings.empty, Strings.empty
        , MediaType.TEXT_X_CUCUMBER_GHERKIN_PLAIN);

  @override
  bool get isEmpty => true;
}
